### PR TITLE
Multi

### PR DESCRIPTION
**Pull request by roman-bytes**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
 node_modules/
+.jshintrc
+dist/
```
Comment (schne324) [inline .gitignore]: we don't want git to ignore dist...especially considering thats what file you use if you npm install this (see "main" of package.json).  The dist directory is where the distributables live

Also, this project uses eslint so  I don't want to randomly have jshintrc in the gitignore

File: index.js [hunk 2]
```diff
@@ -24,6 +24,7 @@ const defaults = {
   input: '.combobox',
   list: '.listbox',
   options: '.listbox .option',
+  optgroup: '.listbox .optgroup[data-label]',
```
Comment (schne324) [inline index.js]: I think opt groups are far too uncommon to have this plugin default to an opt group.  It by default should be turned off and only if a `new Combobox` is instantiated with the `optgroup` configuration option should the functionality be turned on

File: index.js [hunk 3]
```diff
@@ -40,6 +41,7 @@ module.exports = class Combobox {
     this.input = elHandler(this.config.input);
     this.list = elHandler(this.config.list);
     this.cachedOpts = this.currentOpts = elHandler((this.config.options), true);
+    this.cachedOptGrps = elHandler((this.config.optgroup), true);
```
Comment (schne324) [inline index.js]: only set this if an optgroup option is passed in

File: index.js [hunk 4]
```diff
@@ -185,6 +187,7 @@ module.exports = class Combobox {
     // don't let user's functions break stuff
     this.currentOpts = this.currentOpts || [];
     this.updateOpts();
+    this.updateOptsGrp();
```
Comment (schne324) [inline index.js]: only do this if an optgroup option is passed in

File: index.js [hunk 5]
```diff
@@ -210,6 +213,31 @@ module.exports = class Combobox {
     return this;
   }
 
+  updateOptsGrp() {
+    this.cachedOptGrps.forEach((optgrp) => {
+      const groups = optgrp.getElementsByTagName('div');
```
Comment (schne324) [inline index.js]: what if the markup uses a non-div tag? this code won't do what we need it to

File: index.js (hunk 2 above)
Comment (schne324) [reply index.js]: meaning I think here, it should be `optgroup: null` and elsewhere check for it (`if (this.config.optgroup) { // do optgroupy stuff}`)

File: index.js (hunk 5 above)
Comment (roman-bytes) [reply index.js]: `const groups = optgrp.getElementsByClassName('option');` work better?

File: index.js (hunk 2 above)
Comment (roman-bytes) [reply index.js]: Yup i agree, thats a good catch. Will get this added.

**Pull request closed without merging**